test/junit/scala/tools/nsc/backend/jvm/opt/InlinerTest.scala

```diff
@@ -1505,4 +1505,25 @@ class InlinerTest extends ClearAfterClass {
     assertNoIndy(t)
     assertInvoke(t, "C", "C$$$anonfun$1")
   }
+
+  @Test
+  def t9121(): Unit = {
+    val code =
+      """package p1
+        |object Implicits {
+        |  class ScalaObservable(val underlying: Any) extends AnyVal {
+        |    @inline def scMap[R](f: String): Any = f.toRx
+        |  }
+        |  implicit class RichFunction1[T1, R](val f: String) extends AnyVal {
+        |    def toRx: Any = ""
+        |  }
+        |}
+        |import p1.Implicits._
+        |class C {
+        |  def t(): Unit = new ScalaObservable("").scMap("")
+        |}
+      """.stripMargin
+    val c :: _ = compile(code)
```

Comment:
Is a single source unit sufficient to reproduce the issue? I thought it only showed up on separate compilation.